File: app/src/main/java/com/codemybrainsout/imageviewer/view/collection/CollectionContract.java
package com.codemybrainsout.imageviewer.view.collection;

import com.codemybrainsout.imageviewer.model.Collection;
import com.codemybrainsout.imageviewer.model.Photo;
import com.codemybrainsout.imageviewer.model.User;
import com.codemybrainsout.imageviewer.view.base.BasePresenter;
import com.codemybrainsout.imageviewer.view.base.BaseView;
import com.codemybrainsout.imageviewer.view.photo.PhotoContract;

import java.util.List;

/**
 * Created by ahulr on 09-06-2017.
 */

public class CollectionContract {

    interface Presenter extends BasePresenter {

        void openUser(User user);

        void loadCollections(int page, int limit);

        void loadMoreCollections(int page, int limit);

        void openCollection(Collection collection);
    }

    interface View extends BaseView<Presenter> {

        void showUser(User user);

        void showCollection(Collection collection);

        void refreshCollections(List<Collection> list);

        void addCollections(List<Collection> list);

    }

}

File: app/src/main/java/com/codemybrainsout/imageviewer/view/collection/CollectionContract.kt
package com.codemybrainsout.imageviewer.view.collection

import com.codemybrainsout.imageviewer.view.base.BasePresenter
import com.codemybrainsout.imageviewer.view.base.BaseView
import com.codemybrainsout.imageviewer.model.*
import com.codemybrainsout.imageviewer.model.Collection

/**
 * Created by ahulr on 09-06-2017.
 */
class CollectionContract {
    interface Presenter : BasePresenter {
        fun openUser(user: User?)
        fun loadCollections(page: Int, limit: Int)
        fun loadMoreCollections(page: Int, limit: Int)
        fun openCollection(collection: Collection?)
    }

    interface View : BaseView<Presenter?> {
        fun showUser(user: User?)
        fun showCollection(collection: Collection?)
        fun refreshCollections(list: List<Collection>)
        fun addCollections(list: List<Collection>)
    }
}